### ISSUE TITLE
somehow doesn't work ... 

### ISSUE THREAD

**Issue by RensbourroTaki** [LukasKuppers/unity-spline-extrusion#1]
Hi!

I tried to use your spline extrusion scripts, but it somehow doesn't work with the latest Unity 2022.x LTS and Spline Package version 2.5.2
I added the _spline container_, created a spline, added your script, then a _mesh filter_ and _mesh renderer_ but no extrusion is happening. The mesh filter doesn't make sense at all, because it doesn't recognize anything given by the extrusion script.  Btw the axis would not be a problem, because you would see anything anway. However, you have a axis switcher integrated too, but no effect at all. 

I found a different solution at the asset store.

**Comment by RensbourroTaki** [comment]
Update: I tried to add [ExecuteInEditMode] to both scripts, now I see something, but it doesn't look like he's extruding something, also there is no update when manipulating the spline. Sadly I am not a programmer.

**Comment by LukasKuppers** [comment]
Hey, sorry about the late reply.

If you want I can share an example project that uses the script so you can see how it works.
I didn't thoroughly test the script, so there's definitely a chance that it includes a bug.

**Comment by RensbourroTaki** [comment]
Hi Lukas!

Thanks for the offert for the sample scene. I am an experienced unity guy, just not good in programming. Maybe shaders, but that's it then.  I guess I have a pretty good understanding how it works.
It definitely doesn draw the mesh correctly, with the latest Unity 2022.x LTS version.

It's sad, because there is one more other solution in the asset store which I bought, but that system can't handle even the connection of vertex points from one part to the other one. And has even scaling issues by not taking the original mesh 1:1, which causes distorted textures and so on. And then you need to stitch that thing somehow together, and it doesn't fit at all. 15 years ago I worked with a much betters solution, which is today ofc deprecated. However, your solution would be interesting to see in action, as it seems to handle things better, from what I saw. But something tells me you don't want to put any more effort into it anymore. This is something the asset store would really need. And ofc I would need it! Because I want to recreate an old game I did 15 years ago. And my new character controller forces me to use the inegretated Unity's spline solution. So I am totally bound to it.
A video of the old version of the game: https://www.youtube.com/watch?v=H44_WTwcwp0&t=3s

Idk, it would be great if you are willing to make it at least work again? I am not rich, but I would also pay you some bucks for some enhancements, like the cap start & cap end mesh? Well, I don't want to scare you away right away. But I would be happy if your system worked as it is.

[https://i.ytimg.com/vi/H44_WTwcwp0/hqdefault.jpg]<https://www.youtube.com/watch?v=H44_WTwcwp0&t=3s>
The game that was never released:: DYNABOT - The Robo Marble<https://www.youtube.com/watch?v=H44_WTwcwp0&t=3s>
This game was made by a team of two. We worked on it for almost four years. However, you should never go for a project that requires a larger execution team ...
www.youtube.com



________________________________
Von: Lukas Kuppers ***@***.***>
Gesendet: Donnerstag, 4. April 2024 15:48
An: LukasKuppers/unity-spline-extrusion ***@***.***>
Cc: RensbourroTaki ***@***.***>; Author ***@***.***>
Betreff: Re: [LukasKuppers/unity-spline-extrusion] somehow doesn't work ... (Issue #1)


Hey, sorry about the late reply.

If you want I can share an example project that uses the script so you can see how it works.
I didn't thoroughly test the script, so there's definitely a chance that it includes a bug.

**Comment by RensbourroTaki** [comment]
I took another attempt and found it does an extrusion, but only in runtime and if the mesh filter has none before going into runtime. flipping the mesh in 3dsmax & reseting its x-form for proper export still didn't brought the right result. and sadly there is no constant update to play around with the extrusion axis, so I need to go out, change what I want, and then go back in play mode to see if it works. sure it was that way designed not to have constant update in runtime. No matter how I turn and export that thing, it seems not to point in the right direction. xD

**Comment by RensbourroTaki** [comment]
I got it! xD Now the best is.. I rotated it many times again, and came to the same rotation, but this time it worked? xD Idk why! Thank you for this script! It helps me to get along with the development of the game! 

![3dsmaxShape](https://github.com/LukasKuppers/unity-spline-extrusion/assets/123490511/4e4d2a4d-f040-4b23-9e55-add3bc0d46d6)
![splinemesher](https://github.com/LukasKuppers/unity-spline-extrusion/assets/123490511/16db0ad7-cf08-4f8b-bd67-9dcfe35aa23a)

**Comment by RensbourroTaki** [comment]
Ok, the zig zag is happening again with the same mesh! It must be something I am missing, or it is a calulation problem. Because I just tried to replicate the same thing again, and its happening again!

**Comment by RensbourroTaki** [comment]
Please, would you provide me the sample scene? Most interesting would be the exported mesh itself! Because I need to find out the real position of the pivot on the mehs and the total allowed length of the track part. It seems it also depends from which axis the object is drawn. Yes, you told Z, but each programm and export makes something different out of it. It would be very helpful to get your sample scene for inspection. 

Best regards!

**Comment by LukasKuppers** [comment]
Hey,

Glad you got it working!

I threw that script together pretty quickly, so I'm not surprised if it has
some issues haha.

I'm open to improving this, adding more features, and releasing it for free
on the asset store - based on your experience do you think there would be
significant interest in that?

**Comment by RensbourroTaki** [comment]
I'm sorry for deleting all my comments here, I didn't mean to bother you any further. 😬

However, regarding your question:

There is actually no spline mesh extrusion tool in the asset store that doesn't cost something. Most of them even a fortune! 50 to 80,-.  I own them all. 🙄 And only one of them supports Unity's official spline system and that is [spline mesher](https://assetstore.unity.com/packages/tools/modeling/spline-mesher-280289) for Euro 19,-.  And regarding the fact that you want to enhance the tool and give it away for free would be definitely be a reason for the people to download yours. I would just call it something like **_Unity's Spline Custom Mesh Extruder_**, so that it's clear that the Unity spline system is supported! I for example don't like external spline solutions. Some of them are pretty complicated. For example Mega Shapes & Dreamteck Splines, and these are expensive packages. So I'm pretty sure your free and improved solution can lead to a large number of downloads over time.

And I don't know how far you want to improve it, but there is a lot that can be done, and its even that hard for someone like you, I guess. If you need a few tips for improvements, I'm always happy to help. Just let me know.

**Comment by pevas** [comment]
Hi Lukas, I was looking for a script that just adds the extrusion to Unity Splines so the way you did it is perfect. It lets people stick with the core package and just adds a useful functionality on top rather than migrate to a new and complicated 3rd party asset. Unfortunately also for me it is not displaying anything. Would be greatful for the demo project! Thanks.

**Comment by LukasKuppers** [comment]
Did you enter play mode? The script only generates the mesh on Awake, so it won't be visible in edit mode.

**Comment by pevas** [comment]
Yes of course, I also tried different meshes (like tracks and fences) and
settings but to no avail. If the demo scene would be too much work maybe if
you could upload just the example mesh and I'll try with that and report.
Thanks.

**Comment by LukasKuppers** [comment]
I should be able to provide a demo scene in reasonable time. Ill comment here when I do.

**Comment by pevas** [comment]
Any chance for this? At least the mesh that was used would be good to see if I'm just making an error when creating that. Thanks

**Comment by LukasKuppers** [comment]
Hey! Sorry, totally forgot about this!

I have a little prototype project that uses the scirpt: [unity-rts-logging-game](https://github.com/LukasKuppers/unity-rts-logging-game/tree/main)
There's other stuff in there too but if you look around you should find it. Keep in mind that the mesh will only be generated once you enter play mode.
